### ISSUE TITLE
Actorsystem is 'killable'

### ISSUE THREAD

**Issue by pjz** [godaddy/Thespian#5]
Try running:
```python
from thespian.actors import Actor, ActorExitRequest, ActorSystem

class Kill_The_Messenger(Actor):
    def receiveMessage(self, message, sender):
        self.send(sender, ActorExitRequest())

def main():
    asys = ActorSystem()
    ktm = asys.createActor(Kill_The_Messenger)
    asys.tell(ktm, "Test message")
    asys.tell(ktm, ActorExitRequest())

if __name__ == '__main__':
    main()
```

so you can see:

```
Traceback (most recent call last):
  File "buglet.py", line 15, in <module>
    main()
  File "buglet.py", line 12, in main
    asys.tell(ktm, ActorExitRequest())
  File "/tmp/bugl/local/lib/python2.7/site-packages/thespian/actors.py", line 497, in tell
    self._systemBase.tell(actorAddr, msg)
  File "/tmp/bugl/local/lib/python2.7/site-packages/thespian/system/simpleSystemBase.py", line 456, in tell
    self._pendingSends.append(PendingSend(self.actorRegistry['System:ExternalRequester'].address, msg, anActor))
AttributeError: 'NoneType' object has no attribute 'address'
```
...which I believe is caused by the `receiveMessage` of `Kill_The_Messenger` sending an `ActorExitRequest` as a response to the "Test Message" which causes the `ActorSystem` to break internal bookkeeping.

**Comment by kwquick** [comment]
Thank you for the report. I will look into this later today.

**Comment by kwquick** [comment]
Thanks again for the report and the code to exercise the issue.  I have fixed the issue and incorporated your submission into the unit tests: https://github.com/godaddy/Thespian/commit/0be590c6507255b9eaa5033cafcbc4809a039c95

This will be incorporated into the upcoming release (most likely version 2.5.8).

**Comment by pjz** [comment]
Nice! Thanks for the prompt attention! I made one comment on that commit; sorry, I used to do work on code review tools for a living so it's become a bit of a habit.

**Comment by kwquick** [comment]
You are welcome, and please please do review and comment!

**Comment by kwquick** [comment]
Resolved in release 2.5.8 (https://github.com/godaddy/Thespian/releases/tag/thespian-2.5.8).
